app/scripts/metamask-controller.test.js

```diff
@@ -742,257 +802,283 @@ describe('MetaMaskController', () => {
-          'password',
-        );
+        it('should remove the identities when the device is forgotten', async () => {
+          await metamaskController.connectHardware(
+            HardwareDeviceNames.trezor,
+            0,
+          );
+          await metamaskController.unlockHardwareWalletAccount(
+            0,
+            HardwareDeviceNames.trezor,
+          );
+          const hardwareKeyringAccount =
+            metamaskController.keyringController.state.keyrings[1].accounts[0];
 
-        await localMetaMaskController.keyringController.addNewKeyring(
-          'Trezor Hardware',
-          {
-            accounts: ['0x123'],
-          },
-        );
+          await metamaskController.forgetDevice(HardwareDeviceNames.trezor);
 
-        await localMetaMaskController.forgetDevice(HardwareDeviceNames.trezor);
-        const { identities: updatedIdentities } =
-          localMetaMaskController.preferencesController.store.getState();
-        expect(updatedIdentities['0x123']).toBeUndefined();
-      });
+          expect(
+            metamaskController.preferencesController.store.getState()
+              .identities,
+          ).not.toContain(hardwareKeyringAccount);
+        });
 
-      it('should wipe all the keyring info', async () => {
-        await metamaskController
-          .connectHardware(HardwareDeviceNames.trezor, 0)
-          .catch(() => null);
-        await metamaskController.forgetDevice(HardwareDeviceNames.trezor);
-        const keyrings =
-          await metamaskController.keyringController.getKeyringsByType(
-            KeyringType.trezor,
+        it('should wipe all the keyring info', async () => {
+          await metamaskController.connectHardware(
+            HardwareDeviceNames.trezor,
+            0,
           );
 
-        expect(keyrings[0].accounts).toStrictEqual([]);
-        expect(keyrings[0].page).toStrictEqual(0);
-        expect(keyrings[0].isUnlocked()).toStrictEqual(false);
+          await metamaskController.forgetDevice(HardwareDeviceNames.trezor);
+          const keyrings =
+            await metamaskController.keyringController.getKeyringsByType(
+              KeyringType.trezor,
+            );
+
+          expect(keyrings[0].accounts).toStrictEqual([]);
+          expect(keyrings[0].page).toStrictEqual(0);
+          expect(keyrings[0].isUnlocked()).toStrictEqual(false);
+        });
       });
-    });
 
-    describe('unlockHardwareWalletAccount', () => {
-      const accountToUnlock = 10;
-      beforeEach(async () => {
-        await metamaskController.keyringController.createNewVaultAndRestore(
-          'password',
-          TEST_SEED,
-        );
-        jest.spyOn(window, 'open').mockReturnValue();
-        jest
-          .spyOn(
-            metamaskController.keyringController,
-            'addNewAccountForKeyring',
-          )
-          .mockReturnValue('0x123');
+      describe('unlockHardwareWalletAccount', () => {
```

Comment:
All mocks under this `describe` have been removed since it is much more useful to put real code under test (since it is now working)